modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIgniteNestedLoopJoin.java

```diff
@@ -307,55 +285,18 @@ else if (joinType == RIGHT || joinType == JoinRelType.FULL) {
                     ? distribution.function()
                     : DistributionFunction.hash();
 
-                IgniteDistribution outDistr; // TODO distribution multitrait support
```

Comment:
may be this comment informative and we will forgot smth?

Reply:
I don't think so. Different distribution options should be created on derive